scripts/bitTrx.js

```diff
@@ -196,123 +196,25 @@
-				};
-
-				const sig = serializeSig(r, s);
-				sig.push(parseInt(shType, 10));
-
-				return Crypto.util.bytesToHex(sig);
-			} else {
-				return false;
-			}
-		}
-
-		// https://tools.ietf.org/html/rfc6979#section-3.2
-		btrx.deterministicK = function(wif, hash, badrs) {
-			// if r or s were invalid when this function was used in signing,
-			// we do not want to actually compute r, s here for efficiency, so,
-			// we can increment badrs. explained at end of RFC 6979 section 3.2
+			// Parse the private key
+			let strPrivKey = bitjs.wif2privkey(wif);
 
-			// wif is b58check encoded wif privkey.
-			// hash is byte array of transaction digest.
-			// badrs is used only if the k resulted in bad r or s.
+			// Generate low-s deterministic ECDSA signature as per RFC6979
+			// [0] = Uint8Array(sig), [1] = Int(recovery_byte)
+			let arrSig = await nobleSecp256k1.sign(Crypto.util.bytesToHex(hash), strPrivKey.privkey, { canonical: true, recovered: true });
 
-			// some necessary things out of the way for clarity.
-			badrs = badrs || 0;
-			const key = bitjs.wif2privkey(wif);
-			const x = Crypto.util.hexToBytes(key['privkey'])
-			const curve = EllipticCurve.getSECCurveByName("secp256k1");
-			const N = curve.getN();
+			// Concat the Signature with it's recovery byte
+			const bSigFinal = new Uint8Array(arrSig[0].byteLength + 1);
+			writeToUint8(bSigFinal, arrSig[0], 0);
+			bSigFinal[bSigFinal.byteLength - 1] = arrSig[1];
 
-			// Step: a
-			// hash is a byteArray of the message digest. so h1 == hash in our case
-
-			// Step: b
-			let v = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
-
-			// Step: c
-			let k = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
-
-			// Step: d
-			k = Crypto.HMAC(Crypto.SHA256, v.concat([0]).concat(x).concat(hash), k, { asBytes: true });
-
-			// Step: e
-			v = Crypto.HMAC(Crypto.SHA256, v, k, { asBytes: true });
-
-			// Step: f
-			k = Crypto.HMAC(Crypto.SHA256, v.concat([1]).concat(x).concat(hash), k, { asBytes: true });
-
-			// Step: g
-			v = Crypto.HMAC(Crypto.SHA256, v, k, { asBytes: true });
-
-			// Step: h1
-			let T = [];
-
-			// Step: h2 (since we know tlen = qlen, just copy v to T.)
-			v = Crypto.HMAC(Crypto.SHA256, v, k, { asBytes: true });
-			T = v;
-
-			// Step: h3
-			let KBigInt = BigInteger.fromByteArrayUnsigned(T);
-
-			// loop if KBigInt is not in the range of [1, N-1] or if badrs needs incrementing.
-			let i = 0;
-			while (KBigInt.compareTo(N) >= 0 || KBigInt.compareTo(BigInteger.ZERO) <= 0 || i < badrs) {
-				k = Crypto.HMAC(Crypto.SHA256, v.concat([0]), k, { asBytes: true });
-				v = Crypto.HMAC(Crypto.SHA256, v, k, { asBytes: true });
-				v = Crypto.HMAC(Crypto.SHA256, v, k, { asBytes: true });
-				T = v;
-				KBigInt = BigInteger.fromByteArrayUnsigned(T);
-				i++
-			};
-
-			return KBigInt;
-		};
+			return Crypto.util.bytesToHex(bSigFinal);
```

Comment:
I think `return Crypto.util.bytesToHex([...arrSig[0], arrSig[1]]);` would be easier to read, instead of concatenating manually